plugins/parquet/parquetembed.cpp

```diff
@@ -227,13 +227,48 @@ arrow::Status ParquetHelper::openReadFile()
         StringBuffer filename;
         StringBuffer path;
         splitFilename(location.c_str(), nullptr, &path, &filename, nullptr, false);
-        Owned<IDirectoryIterator> itr = createDirectoryIterator(path.str(), filename.append("*.parquet"));
+        Owned<IDirectoryIterator> fileItr;
+
+        // hThor and Thor files that were written with parquet have diffent naming schemes
+        // Thor files are tagged with a worker ID while hThor files are single non-tagged files
+        if (activityCtx->numSlaves() > 1)
+        {
+            // Executing on Thor: Check for single files that were written by hThor or from external source
+            Owned<IDirectoryIterator> singleItr = createDirectoryIterator(path.str(), filename.append(".parquet"));
+            if (!singleItr || !singleItr->first())
+            {
+                // Check for partitioned files which will have indexes prefixed to the filename
+                Owned<IDirectoryIterator> multItr = createDirectoryIterator(path.str(), filename.insert(filename.length() - 8, '*'));
+                fileItr = multItr;
+            }
+            else
+            {
+                // There was a single file so check for partitioned files
+                Owned<IDirectoryIterator> multItr = createDirectoryIterator(path.str(), filename.insert(filename.length() - 8, "[0-9]*"));
```

Comment:
I don't think this does what you expect.  "[0-9]\*" is not treated as a regular expression it is a filename wildcard containing '\*' or '?'.  I think you will need to check for filename* and then post-process.  Or use "filename*.parquet"?